src/Gui/DlgGeneralImp.cpp

```diff
@@ -371,6 +393,100 @@ void DlgGeneralImp::changeEvent(QEvent *event)
     }
 }
 
+///////////////////////////////////////////////////////////
+namespace {
+

```

Comment:
This chunk of code is too opaque to be uncommented -- I had to read through it several times to parse what was going on, and I'm still not sure I completely understand all the nuances. The use of `applyDockWidget` and `applyStyleSheet` to do two completely different things depending on `delayTrigger` in particular is quite confusing -- why use the same functions to cover two completely independent code paths?